doppelganger/doppelganger_state.go

```diff
@@ -0,0 +1,28 @@
+package doppelganger
+
+// doppelgangerState tracks the validator's state in Doppelganger Protection.
+type doppelgangerState struct {
+	remainingEpochs uint64 // The number of epochs that must be checked before it's considered safe.
```

Comment:
Like I commented on the const states, I think this could be just `int8` since only options are 1/2 and invalid (imo -1)